pdf2remarkable.sh

```diff
@@ -70,28 +70,74 @@ REMARKABLE_HOST=${REMARKABLE_HOST:-remarkable}
 REMARKABLE_XOCHITL_DIR=${REMARKABLE_XOCHITL_DIR:-.local/share/remarkable/xochitl/}
 TARGET_DIR="${REMARKABLE_HOST}:${REMARKABLE_XOCHITL_DIR}"
 
+# Function to show help
+show_help ()
+{
+    echo "Transfer PDF or EPUB document(s) to a reMarkable tablet."
+    echo "usage: $(basename $0) [options] path-to-file [path-to-file]..."
+    echo "  -h  --help             print this usage information and exit"
+    echo "  -q  --quiet            don't print progress information"
+    echo "  -r  --toggle-restart   toggle whether to restart the tablet (default is given by RESTART_XOCHITL_DEFAULT)"
+    echo "      --uuids-file       write the list of uploaded files and their UUIDs to a file"
+    echo "See also comments/documentation at start of the script."
+}
+
 # Check if we have something to do
 if [ $# -lt 1 ]; then
-    echo "Transfer PDF or EPUB document(s) to a reMarkable tablet."
-    echo "See comments/documentation at start of script."
-    echo "usage: $(basename $0) [ -r ] path-to-file [path-to-file]..."
+    show_help
     exit 1
 fi
 
 RESTART_XOCHITL_DEFAULT=${RESTART_XOCHITL_DEFAULT:-0}
 RESTART_XOCHITL=${RESTART_XOCHITL_DEFAULT}
-if [ "$1" = "-r" ] ; then
-    shift
-    if [ $RESTART_XOCHITL_DEFAULT -eq 0 ] ; then
-        echo Switching
-        RESTART_XOCHITL=1
-    else
-        RESTART_XOCHITL=0
+
+BE_QUIET=0
+SCP_OPTIONS=
+UUIDS_FILE=
+
+# Print progess information
+log () {
+    if  [ $BE_QUIET -eq 0 ]; then
+        echo "$@"
     fi
-fi
+}
+
+
+# Parse arguments
```

Comment:
would using getopt(1) be easier?  or are there linux/osx compatibility issues?

Reply:
I was not aware of getopt(1). Looking at the docs now, it indeed could have saved a little bit of work. I will not attempt change to getopt(1), as it does not seem to offer huge advantages. However, I will keep it in mind when I write argument-parsing next time.